cmd/root.go

```diff
@@ -223,15 +226,45 @@ func parseToolsFile(ctx context.Context, raw []byte) (ToolsFile, error) {
 	return toolsFile, nil
 }
 
-func parse(ctx context.Context, buf []byte, logger log.Logger) {
-	logger.DebugContext(ctx, "Attempting to parse updated tools file.")
-	// time.Sleep(5 * time.Second)
-	// logger.DebugContext(ctx, "Parse finished.")
-	// TODO: add logic for parsing
+func validateReloadEdits(
+	ctx context.Context, toolsFile ToolsFile,
+) (map[string]sources.Source, map[string]auth.AuthService, map[string]tools.Tool, map[string]tools.Toolset, error,
+) {
+	logger, err := util.LoggerFromContext(ctx)
+	if err != nil {
+		panic("attempted to parse logger when not in context")
```

Comment:
```suggestion
		panic(err)
```
since the error that was return is pretty straight forward, we can just use it.